lib/vector-api-client/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 mod client;
```

Comment:
Since this is a crate now, how about adding some crate annotations?

```rust
#![deny(
    missing_debug_implementations,
    missing_copy_implementations,
    missing_docs
)]
```

They will help find spots that can be improved and help with maintenance later on.

Reply:
I think you just like causing me work 😛 

```
error: aborting due to 44 previous errors
```

Adding docs and a bucket load of `#[derive]`s now... 😄

Reply:
:smile: "How to get 44 new compile errors with a neat simple trick"

Reply:
Added docs, and the derived traits in 97e1add9558e270e09a62c91709ab801edf11329.

I've temporarily removed the `missing_docs` rule due to errors like this:

```
error: missing documentation for a struct field
  --> lib/vector-api-client/src/gql/metrics.rs:18:10
   |
18 | #[derive(GraphQLQuery, Debug, Copy, Clone)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

This looks like something `GraphQLQuery` needs to accommodate in its macros -- is that right?